crates/sui-node/src/lib.rs

```diff
@@ -642,18 +634,31 @@ impl SuiNode {
     /// epoch has changed. Upon receiving such signal, we reconfigure the entire system.
     pub async fn monitor_reconfiguration(self: Arc<Self>) -> Result<()> {
         loop {
+            let checkpoint_executor = CheckpointExecutor::new(
+                self.state_sync.subscribe_to_synced_checkpoints(),
+                self.checkpoint_store.clone(),
+                self.state.clone(),
+                self.config.checkpoint_executor_config.clone(),
+                &self.registry_service.default_registry(),
+            )?;
+
+            let executor_handle =
+                spawn_monitored_task!(async move { checkpoint_executor.run().await });
+            executor_handle
+                .await
+                .expect("Failed to join on checkpoint executor run task");
```

Comment:
if we're just spawning a task and then awaiting on the handle, there's no need for it to be in a task.

Reply:
Agreed, as mentioned offline, the main reason I was thinking we keep this around is for leveraging the monitoring we get from `spawn_monitored_task`. But I think the main thing we would want to monitor is the execution tasks, which are still separate tasks. Will remove.